src/NewTools-Debugger-Tests/StDebuggerTest.class.st

```diff
@@ -752,6 +759,56 @@ StDebuggerTest >> testNewDebuggerContextFor [
 	self assert: debuggerContext context identicalTo: ctx
 ]
 
+{ #category : 'tests - context inspector' }
+StDebuggerTest >> testPreserveLastKnownExtent [
+
+	| newDbgExtent lastUserExtent |
+
+	dbg := (self debuggerOn: session) open.
+	"First we save the user extent (test running should not modify user changes)"
+	lastUserExtent := self settings lastKnownExtent.
+	newDbgExtent := 100 @ 200.
+	"Programmatically change the extent of debugger window"
+	dbg window window extent: newDbgExtent.
+	"New extent is saved when the window is closed"
+	dbg close.
+	self tearDown.
+
+	"Now we open the debugger again"
+	self setUp.
+	dbg := (self debuggerOn: session) open.
+	"Check the debugger window is set to the new extent"
+	self assert: dbg window window extent equals: newDbgExtent.
+	"Set back the last extent regardless of this test"
+	dbg window window extent: lastUserExtent.
+
+]
+
+{ #category : 'tests - context inspector' }
+StDebuggerTest >> testPreserveLastKnownPosition [
+
+	| newDbgPosition lastUserPosition |
+
+	dbg := (self debuggerOn: session) open.
+	"First we save the user position (test running should not modify user changes)"
+	lastUserPosition := self settings lastKnownPosition.
+	newDbgPosition := 500 @ 100.
+	"Programmatically change the position of debugger window"
+	dbg window window position: newDbgPosition.
+	"New position is saved when the window is closed"
+	dbg close.
+	self tearDown.
+
+	"Now we open the debugger again"
+	self setUp.
+	dbg := (self debuggerOn: session) open.
+	"Check the debugger window is set to the new position"
+	self assert: dbg window window position equals: newDbgPosition.
+	"Set back the last position regardless of this test"
+	dbg window window position: lastUserPosition.
```

Comment:
maybe we should add tests that test the behavior when `StDebuggerSettings class>>#PreserveWindowPositionAndExtent` is false